File: src/Portfolio.Domain/ValueObjects/EnrichedAssetHolding.cs
namespace Portfolio.Domain.ValueObjects;

public class EnrichedAssetHolding
{
    public string AssetId { get; set; } = null!;
    public decimal Quantity { get; set; }
    public decimal CurrentPrice { get; set; }
    public decimal CurrentValue { get; set; }
    public decimal AvgCost { get; set; }
    public decimal TotalCostBasis { get; set; }
    public decimal CostBasisOfSold { get; set; }
    public decimal OpenPL { get; set; }
    public decimal OpenReturn { get; set; }
    public decimal RealizedPL { get; set; }
    public decimal RealizedReturn { get; set; }
    public decimal TotalPL { get; set; }
    public decimal TotalReturn { get; set; }
    public decimal AllocationPercentage { get; set; }
}

File: src/Portfolio.Client/src/app/models/enriched-asset-holding.ts
export interface EnrichedAssetHolding {
    id: string;
    symbol: string;
    name: string;
    imageUrl?: string;
    quantity: number;
    currentPrice: number;
    currentValue: number;
    avgCost: number;
    totalCostBasis: number;
    costBasisOfSold: number;
    openPL: number;
    openReturn: number;
    realizedPL: number;
    realizedReturn: number;
    totalPL: number;
    totalReturn: number;
    allocationPercentage: number;
}
